docs/plugins/ingest.asciidoc

```diff
@@ -101,9 +87,7 @@ Throws error when the field is not an array.
 --------------------------------------------------
 {
   "join": {
-    "fields": {
-      "joined_array_field": "other_array_field"
-    }
+    "joined_array_field": "-"
```

Comment:
at this point I wonder if we should always require a field and a pattern for these processors too? key:value might be more verbose but clearer. it would make things also more consistent with other processors.